lms/djangoapps/grades/signals/handlers.py

```diff
@@ -241,10 +245,9 @@ def recalculate_course_and_subsection_grades(sender, user, course_key, **kwargs)
     Updates a saved course grade, forcing the subsection grades
     from which it is calculated to update along the way.
     """
-    previous_course_grade = CourseGradeFactory().read(user, course_key=course_key)
-    if previous_course_grade and previous_course_grade.attempted:
```

Comment:
Nit question: How often do we expect enrollment and cohort membership changes for user who have yet to submit a problem?  I assume not very often.  If it becomes an issue, we could check for `attempted` both within the task and before creating the task.